src/uart.rs

```diff
@@ -559,10 +696,12 @@ impl<'d> UartDriver<'d> {
         self.tx().write(buf)
     }
 
+    /// Clears the receive buffer.
     pub fn flush_read(&self) -> Result<(), EspError> {
         self.rx().flush()
     }
 
+    /// Waits for the transmission to complete.
     pub fn flush_write(&self) -> Result<(), EspError> {
```

Comment:
I find the naming here quite confusing tbh, I would expect `flush_write` to clear the transmit buffer, but that is a functionality which is not available (apparently due to a hardware bug).

I think the documentation improves the situation, but I could also deprecate the functions and create new ones with more self-describing naming. Any thoughts?